server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -896,6 +896,8 @@ public void evaluateRequestShouldUseConcurrentSearch() {
             && aggregations().factories() != null
             && !aggregations().factories().allFactoriesSupportConcurrentSearch()) {
                 requestShouldUseConcurrentSearch.set(false);
+            } else if (terminateAfter != DEFAULT_TERMINATE_AFTER) {
+                requestShouldUseConcurrentSearch.set(false);
```

Comment:
If we disable concurrent search here, while there is a timeout in effect, this can have the effect of causing a significant reduction in total documents scanned across all nodes. That means you would get different results.

Reply:
> If we disable concurrent search here, while there is a timeout in effect,

I am not sure I understand the issue, but this is how aggregations work right now (they are sequential, not concurrent). The concurrent path have some limitation related to timeouts though, that is correct

Reply:
I'm also not sure I understand the issue. What is happening here is that we are reverting back to the non-concurrent search workflow whenever the `terminate_after` search parameter is used.

Here is a (very long) issue with more details on the problems surrounding `terminate_after`, force termination, and concurrent search: https://github.com/opensearch-project/OpenSearch/issues/8371